src/distributors/adoptopenjdk-installer.ts

```diff
@@ -0,0 +1,79 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+
+import { IS_WINDOWS, PLATFORM, IS_MACOS, macOSJavaContentDir } from "../util";
+import { JavaInstallerResults, IJavaRelease, JavaBase, JavaInstallerOptions } from "./base-installer";
+import { IRelease, IReleaseVersion } from './adoptopenjdk-models'
+
+export class AdoptOpenJdkDistributor extends JavaBase {
+    
+    constructor(initOptions: JavaInstallerOptions) {
+        super("AdoptOpenJDK", initOptions);
+    }
+
+    private async getAvailableMajor(range: semver.Range) {
+        const availableMajorVersionsUrl = "https://api.adoptopenjdk.net/v3/info/available_releases"
+        const availableMajorVersions = (await this.http.getJson<IReleaseVersion>(availableMajorVersionsUrl)).result;
+
+        if (!availableMajorVersions) {
+            throw new Error(`No versions were found for ${this.distributor}`)
+        }
+
+        const coercedAvailableVersions = availableMajorVersions.available_releases.map(item => semver.coerce(item)!)!;
```

Comment:
`.map(item => semver.coerce(item)!)!;` looks a bit strange.
May be `.map(item => semver.coerce(item)).filter(item => Boolean(item))`

Reply:
Done.